www/canvas/sherlock/googlePredictionHelpers.js

```
@@ -0,0 +1,100 @@
+              googleQueryModels.push(predictedScore);
+            }
+          }
+        });
+      resolve({googleQueryModels,sortedQuestions});
+    });
+  });
+};
+
+//constructs comments for questions and creates grade modifier for assignment score
+Predict.processGooglePredictionScoring = function(googleGradesAndSortedQuestions){
+  return new Promise(function(resolve,reject){
+    Promise.all(googleGradesAndSortedQuestions.googleQueryModels)
+      .then(function(scoreData){
+        const sherlockComments = {};
+        const sortedQuestions = googleGradesAndSortedQuestions.sortedQuestions;
+        const scoreModifiers = scoreData.map(function(score){
+          const q = score.question;
+          const questionNum = q.title.match(/\d{2}/)[0];
+          const idx = score.sortedQuestionIdx;
+          const typeformScore = sortedQuestions[idx+1] ? Grade.typeformGrade(questionNum, sortedQuestions[idx+1].title) : 1;
+          const predictedScore = score.score;
+          const adjustment = predictedScore - typeformScore;
+          if(adjustment !== 0) {
+            const adjText = adjustment > 0 ? `+${adjustment}` : adjustment;
+            sherlockComments[q.id] = `Sentiment Analysis ${adjText}`;
+          }
+          return adjustment;
+        });
+        const gradeModifier = scoreModifiers.reduce(function(a,b){return a+b;}, 0);
+        resolve({gradeModifier, sherlockComments});
+      });
+  });
+};
+
+//gets the list of prediction models
+function getModelList(){
+  return new Promise(function(resolve,reject){
+    predictionClient.getModels((err, models) => {
+      if (err) CONSOLE_WITH_TIME(err);
+      // `models` is an array of Model objects.
+      if(!models){
+        //added to bypass Google Prediction if API call for models fails
+        CONSOLE_WITH_TIME('WARNING: Call to get models from Google Prediction failed');
+        CONSOLE_WITH_TIME('Reverting to using normal Typeform grading.');
+        resolve([]);
```

Do we want to reject here instead of resolving empty?